### PR TITLE
Windows - distinguishing between call and shell for better arg quoting

### PR DESCRIPTION
**Pull request by joerick**
Alternate version of #399  :)

I had a slightly different idea to @henryiii on this one and wanted to try it. Let's see what CI makes of it.

### CODE REVIEW

**Comment by joerick** [comment]
Cool, looks like this is working. Does this look okay to you @YannickJadoul @henryiii ? I can get it merged and in a release pretty soon if so.

**Comment by henryiii** [comment]
I think it's good!

**Comment by YannickJadoul** [comment]
A bit sad this doesn't work the same as macos, then:

https://github.com/joerick/cibuildwheel/blob/d0b6401129ead1164d012835ea6366c8486a562e/cibuildwheel/macos.py#L17-L24

But a similar `call([repair_command_prepared], env=env)` doesn't work, right?

File: test/test_pep518.py [hunk 1]
```diff
@@ -24,7 +28,8 @@
 requires = [
     "setuptools>=42",
```
Comment (henryiii) [inline test/test_pep518.py:29]:
```suggestion
    "setuptools >= 42",
```

We could add this, just to ensure this covers #392

**Comment by YannickJadoul** [comment]
(+1 on the newly-coined terminology of "the shell/call debacle", btw! ;-) )

**Pull request merged**